torch/_functorch/make_functional.py

```diff
@@ -497,6 +497,54 @@ def combine_state_for_ensemble(models):
     return funcs[0], params, buffers
 
 
+def combine_weights_for_ensemble(models):
+    """combine_weights_for_ensemble(models) -> params, buffers
```

Comment:
> This includes params and buffers, and "weights" usually just means "params", so shouldn't the name for this be combine_state_for_ensemble? (lol that is what the old one was called)

Yeah in general I was wondering if that was going to make it confusing for users moving from `functorch` to `torch.func` since the signature is going to be different (no function returned). Since IMO if I'm doing that change, I'm doing CMD + F of "functorch" => "torch.func" and then cleaning up the functions that don't exist (mostly make_functional and depending on this combine_state_for_ensemble)

Mostly I was thinking that having the function not exist under the same name as in functorch would make it easier for an existing user to see the changes, but happy to change it back to combine_state_for_ensemble

Reply:
> Yeah in general I was wondering if that was going to make it confusing for users moving from functorch to torch.func since the signature is going to be different (no function returned).

That's a good point. My problem with the name `combine_weights_for_ensemble` is that the implication is that it only combines parameters and does nothing with the buffers

Should we try to come up with another name for this?

Reply:
> Should we try to come up with another name for this?

Sure! Spitballing some ideas here:
- combine_params_and_buffers_for_ensemble (maybe too wordy)
- ensembled_state (maybe too close, but might at least be different enough for someone to look at the signature? similar to make_functional => functional_call)
- ensembled_params_and_buffers
- concat_state_for_ensemble (also maybe too close but accurately describes that we are only concating)

Reply:
some more ideas after reading yours:
- `stack_module_state` (NB: it's a torch.stack more than a torch.cat)
- `ensembled_module_state` (makes it clearer that it operates on modules, and not e.g. optimizers)

I agree that `combine_params_and_buffers_for_ensemble` is wordy, I could be happy with any of the other names

Reply:
Hmm, so an "ensemble of modules" doesn't necessarily need to have all of its parameters/buffers stacked together. The for-loop approach is still a valid approach. So having stack or concat in the API name might help

Reply:
`stack_module_state` makes sense to me! I'll change it